### PR TITLE
Small changes to Kotlin Koans and Learning materials overview pages

### PR DESCRIPTION
**Pull request by sarahhaggarty**
This PR suggests improvements to:
- Explain that some Java experience is helpful for completing Kotlin Koans.
- Add a reference for readers that want to publish a multiplatform library to a GitHub repository (KWHF-812).

There are also other smaller improvements in other areas related to word choice.

### CODE REVIEW

File: docs/topics/learning-materials-overview.md [hunk 1]
```diff
@@ -6,12 +6,12 @@ You can use the following materials and resources for learning Kotlin:
...
-* [Advent of Code puzzles](advent-of-code.md) – learn idiomatic Kotlin and practice your language skills by completing short and fun tasks.
+* [Kotlin tips](kotlin-tips.md) – watch short videos where the Kotlin team show you how to use Kotlin in a more efficient and idiomatic way, so you can have more fun when writing code.
+* [Advent of Code puzzles](advent-of-code.md) – learn idiomatic Kotlin and test your language skills by completing short and fun tasks.
 * [Kotlin hands-on tutorials](https://play.kotlinlang.org/hands-on/overview) – complete long-form tutorials to fully grasp a technology. These tutorials guide you through a self-contained project related to a specific topic.
-* [Kotlin for Java Developers](https://www.coursera.org/learn/kotlin-for-java-developers) – course for developers with experience in Java. It shows the similarities between the two languages and focuses on what's going to be different.
-* [Kotlin documentation in the PDF format](kotlin-pdf.md) – read the whole documentation offline.
+* [Kotlin for Java Developers](https://www.coursera.org/learn/kotlin-for-java-developers) – learn the similarities and differences between Java and Kotlin.
```
Comment (koshachy) [inline docs/topics/learning-materials-overview.md:16]: Hmmm 🤔 

It's not clear now that it is a course, but this is important info IMO. Users should click on the link to understand that this is a course, not an article about differences between Java and Kotlin.

I'd like you to mention the `course` term explicitly here.

File: docs/topics/learning-materials-overview.md [hunk 2]
```diff
@@ -6,12 +6,12 @@ You can use the following materials and resources for learning Kotlin:
...
+* [Kotlin documentation in the PDF format](kotlin-pdf.md) – read all our documentation offline.
```
Comment (koshachy) [inline docs/topics/learning-materials-overview.md:17]: PDF doesn't contain API references :( 
So shouldn't we omit "all our" from the sentence?

**Review by koshachy — CHANGES_REQUESTED** [review]
Nice!
Left some comments. If you have any questions, feel free to reach out to me to discuss them.

File: docs/topics/learning-materials-overview.md (hunk 2 above)
Comment (sarahhaggarty) [reply docs/topics/learning-materials-overview.md:17]: What about: "read Kotlin documents offline"? An API reference is a form of "documentation" so using "documents" instead would narrow the scope. Or we could explicitly mention that API references are not included.

File: docs/topics/learning-materials-overview.md (hunk 1 above)
Comment (sarahhaggarty) [reply docs/topics/learning-materials-overview.md]: Good point! Can you check my updated proposal?

Comment (koshachy) [reply docs/topics/learning-materials-overview.md]: Works for me

File: docs/topics/learning-materials-overview.md (hunk 2 above)
Comment (koshachy) [reply docs/topics/learning-materials-overview.md]: My concern was only about "all".
_read all Kotlin documentation offline, excluding API references_ (could be rephrased) also works

Comment (sarahhaggarty) [reply docs/topics/learning-materials-overview.md]: Ah. I get it now! I've removed "all". I decided not to mention the exclusion of API references as it is clearly explained in kotlin-pdf.md.

**Review by koshachy — APPROVED** [review]

**Review by PetrakovichVictoria — APPROVED** [review]

File: docs/topics/koans.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [//]: # (title: Kotlin Koans)
 
-Kotlin Koans are a series of exercises to get you familiar with the Kotlin syntax.
+Kotlin Koans are a series of exercises designed primarily for Java developers, to help you become familiar with the Kotlin syntax.
```
[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/koans.md:3]: (Following our talk in Slack)

File: docs/topics/learning-materials-overview.md [hunk 4]
```diff
@@ -6,12 +6,12 @@ You can use the following materials and resources for learning Kotlin:
   * [Java to Kotlin migration guide: Strings](java-to-kotlin-idioms-strings.md) – learn how to perform typical tasks with strings in Java and Kotlin.
   * [Java to Kotlin migration guide: Collections](java-to-kotlin-collections-guide.md) — learn how to perform typical tasks with collections in Java and Kotlin.
   * [Java to Kotlin migration guide: Nullability](java-to-kotlin-nullability-guide.md) — learn how to handle nullability in Java and Kotlin.
-* [Kotlin Koans](koans.md) – complete exercises to learn the Kotlin syntax. Each exercise is created as a failing unit test and your job is to make it pass.
+* [Kotlin Koans](koans.md) – complete exercises to learn the Kotlin syntax. Each exercise is created as a failing unit test and your job is to make it pass. Recommended for developers with Java experience.
```
[the remark this answers is not in the document]
Comment (sarahhaggarty) [reply docs/topics/learning-materials-overview.md:9]: Hmmm I see what you mean. It comes across firmer when it's active. But we don't want to make it seem like Java developers **have** to complete Koans 🤔 I think the original is better in this specific case, even though it is passive 😅 Is it OK with you if I don't change it?

[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/topics/learning-materials-overview.md:9]: Sure! Just wanted to show you what I meant (in DM in Slack).

@koshachy Do I understand correctly that you mean it's not necessary to write "we recommend"?

[the remark this answers is not in the document]
Comment (koshachy) [reply docs/topics/learning-materials-overview.md:9]: We usually use "we" when we need to refer to the Kotlin team and highlight that we have done something. That's why we use a lot of it in the "What's new" articles and almost avoid using it in others. 
So we shouldn't abuse this pronoun. :) Not quite sure that this case requires using it.

[the remark this answers is not in the document]
Comment (koshachy) [reply docs/topics/learning-materials-overview.md:9]: If I'd like to spend more time here with this small PR and omit "we", I'd rephrase it to smth like:
"It is a quick start for developers with Java experience." 
"These exercises require Java programming experience, but also work if you got any other programming language experience". 
"To start with Koans, you should have (it's better to have) programming experience in Java or any other language."

and a lot of other variations.
